app/registries/foreman/plugin.rb

```diff
@@ -39,6 +40,7 @@ class PluginRequirementError < Foreman::Exception; end
   class Plugin
     @registered_plugins = {}
     @tests_to_skip = {}
+    @fact_registry = Plugin::ImporterRegistry.new
```

Comment:
This is a bit confusing - is this a fact registry or an importer registry? I would recommend naming the class and the variable wit the same name.